java-persistence/src/main/java/com/whitecloak/training/inventory/controller/ItemController.java

```diff
@@ -19,10 +24,36 @@ public ItemController(ItemService itemService) {
         this.itemService = itemService;
     }
 
-    @PostMapping("/api/v1/items")
+    @PostMapping
     public ItemResource createItem(@RequestBody ItemForm form) {
         return itemService.create(form);
     }
 
-    @GetMapping("/api/v1/items/{id}")
+    @ResponseStatus()
+    @GetMapping("/{id}")
+    public ItemResource getItem(@PathVariable("id") Long id){
+        try {
+            return itemService.getItem(id);
+        }catch (NullPointerException ex){
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Item not found", ex);
+        }
+
+    }
+
+    @GetMapping
+    public Page<ItemResource> getItems(@RequestParam(name = "category", required = false) String category, Pageable pageable){
+        if(category == null){
+            return itemService.getItems(pageable);
+        }
+        else{
+            return itemService.getItems(category, pageable);
+        }
+    }
+
+    @DeleteMapping("/{id}")
+    public ItemResource deleteItem(@PathVariable("id") Long id){
+        itemService.deleteItem(id);
+        return null;
```

Comment:
Might as well set this method to return `void` when the method is returning `null` 100% of the time